unit_test.py

```diff
@@ -19,70 +31,60 @@ def test_link_detect_finds_multiple_links():
 
+def test_valid_link_inserts_prefix_if_none(monkeypatch):
+    monkeypatch.setattr(shownoter, 'get', mock_get)
 
-def test_valid_link_inserts_prefix_if_none():
-    assert shownoter.valid_link('link.com') == 'http://link.com'  
+    link = shownoter.Link('link.com')
+    assert 'http://link.com' == link.url
 
-def test_valid_link_does_nothing_if_prefix_exists():
-    assert shownoter.valid_link('http://link.com') == 'http://link.com'
+def test_valid_link_does_nothing_if_prefix_exists(monkeypatch):
+    monkeypatch.setattr(shownoter, 'get', mock_get)
 
-@pytest.fixture
-def mock_html():
-    return '<html><head><title>Test</title></head></html>'
+    link = shownoter.Link('http://link.com')
+    assert 'http://link.com' == link.url
 
 @requests_mock.Mocker(kw='mock')
 def test_title(mock_html, **kwargs):
-    link = 'http://link.com' 
+    link = 'http://link.com'
     html = kwargs['mock'].get(link, text=mock_html)
     title = shownoter.title(link)
     assert html.called
     assert title == "Test"
 
-def test_markdown():
+def test_link_markdown():
     link = 'link.com'
     title = 'Test'
-    assert shownoter.markdown(link, title) == '[Test](link.com)'
+    assert '[Test](link.com)' in shownoter.link_markdown(title, link)
 
-def test_markdown_if_image():
+def test_image_markdown():
     link = 'link.png'
     title = ''
-    assert shownoter.markdown(link, title) == '![](link.png)'
+    assert '![](link.png)' in shownoter.image_markdown(title, link)
 
-def test_is_image_detect_png():
+def test_image_detect_detects_png():
     link = 'link.png'
-    assert shownoter.is_image(link)
-    
-def test_is_image_detect_jpg():
+    assert shownoter.image_detect(link)
+
+def test_image_detect_detects_jpg():
     link = 'link.jpg'
-    assert shownoter.is_image(link)
+    assert shownoter.image_detect(link)
 
-def test_is_image_detect_gif():
+def test_image_detect_detects_gif():
     link = 'link.gif'
-    assert shownoter.is_image(link)
+    assert shownoter.image_detect(link)
 
-def test_is_image_does_not_detect_outside_other_links():
+def test_image_detect_does_not_detect_outside_other_links():
     link = 'link.foo'
-    assert not shownoter.is_image(link)
+    assert not shownoter.image_detect(link)
 
 @requests_mock.Mocker(kw='mock')
 def test_title(mock_html, **kwargs):
-    link = 'http://link.com' 
+    link = 'http://link.com'
     html = kwargs['mock'].get(link, text=mock_html)
-    sample_link = shownoter.link(link)
+    sample_link = shownoter.Link(link)
     assert html.called
-    assert sample_link['url'] == 'http://link.com'
-    assert sample_link['title'] == 'Test'
-    assert sample_link['markdown'] == '[Test](http://link.com)'
-
-def test_shownotes():
-    description = 'This is a test'
-    links = ['foo', 'bar']
-    notes = shownotes(description=description, links=links)
-    assert notes == description + '\n\nfoo\nbar\n'
-
-def test_links_to_string():
```

Comment:
I actually put this function back into the code. If you want you can do a pull in a couple minutes and the function will be in again `show noter.links_to_string()`